api/src/test/java/org/openmrs/scheduler/tasks/AutoRetireUsersTaskTest.java

```diff
@@ -0,0 +1,161 @@
+ */
+package org.openmrs.scheduler.tasks;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.openmrs.GlobalProperty;
+import org.openmrs.User;
+import org.openmrs.api.AdministrationService;
+import org.openmrs.api.UserService;
+import org.openmrs.api.context.Context;
+import org.openmrs.test.jupiter.BaseContextSensitiveTest;
+import org.openmrs.util.OpenmrsConstants;
+
+import java.util.Date;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+class AutoRetireUsersTaskTest extends BaseContextSensitiveTest {
+	private static final long ONE_DAY_IN_MILLISECONDS = 24 * 60 * 60 * 1000;
+	private static final long TWENTY_THREE_HOURS_IN_MILLISECONDS = 23 * 60 * 60 * 1000;
+	private static final long TWO_DAYS_IN_MILLISECONDS = 2 * ONE_DAY_IN_MILLISECONDS;
+	private static final String ONE_DAY_PROPERTY_VALUE = "1";
+	private static final String TWO_DAYS_PROPERTY_VALUE = "2";
+	private static final String AUTO_RETIRE_REASON = "User retired due to inactivity";
+
+	private UserService userService;
+	private AdministrationService administrationService;
+	private AutoRetireUsersTask autoRetireUsersTask;
+
+	@BeforeEach
+	public void setup() {
+		administrationService = Context.getAdministrationService();
+		autoRetireUsersTask = new AutoRetireUsersTask();
+		userService = Context.getUserService();
+	}
+
+	@Test
+	public void shouldRetireUsersWhoseInactivityExceedNumberOfDaysToRetire() {
+		administrationService.saveGlobalProperty(new GlobalProperty(OpenmrsConstants.GP_NUMBER_OF_DAYS_TO_AUTO_RETIRE_USERS, ONE_DAY_PROPERTY_VALUE));
+
+		User user = getDefaultUser();
+		userService.setUserProperty(
+			user,
+			OpenmrsConstants.USER_PROPERTY_LAST_LOGIN_TIMESTAMP,
+			String.valueOf((System.currentTimeMillis() - TWO_DAYS_IN_MILLISECONDS))
+		);
+
+		user.getAllRoles().forEach(user::removeRole);
+
+		autoRetireUsersTask.execute();
+
+		assertTrue(user.isRetired());
+		assertEquals(user.getRetireReason(), AUTO_RETIRE_REASON);
+	}
+
+	@Test
+	public void shouldNotRetireUsersWhoseInactivityDoNotExceedNumberOfDaysToRetire() {
+		administrationService.saveGlobalProperty(new GlobalProperty(OpenmrsConstants.GP_NUMBER_OF_DAYS_TO_AUTO_RETIRE_USERS, TWO_DAYS_PROPERTY_VALUE));
+
+		User user = getDefaultUser();
+		userService.setUserProperty(
+			user,
+			OpenmrsConstants.USER_PROPERTY_LAST_LOGIN_TIMESTAMP,
+			String.valueOf((System.currentTimeMillis() - ONE_DAY_IN_MILLISECONDS))
+		);
+
+		user.getAllRoles().forEach(user::removeRole);
+
+		autoRetireUsersTask.execute();
+
+		assertFalse(user.isRetired());
+	}
+
+	@Test
+	public void shouldNotRetireAlreadyRetiredUsers() {
+		administrationService.saveGlobalProperty(new GlobalProperty(OpenmrsConstants.GP_NUMBER_OF_DAYS_TO_AUTO_RETIRE_USERS, ONE_DAY_PROPERTY_VALUE));
+
+		String retireReason = "Retire Test users";
```

Comment:
Shouldn't this be a constant?

Reply:
Not really. This is because the variable's scope in within this method.

Reply:
I mean a constant within the method.